packages/vue-components/src/Overlay.vue

```diff
@@ -0,0 +1,185 @@
+<template>
+  <div>
+    <span
+      :id="buttonId"
+      :class="['glyphicon', { 'nav-menu-close-icon': show }]"
+      @click="toggleNavMenu"
+    ></span>
+    <div ref="navMenuContainer" :class="[navMenuIconClass, { 'nav-menu-open': show }]">
+      <retriever
+        v-if="hasIdentifier"
+        :src="src"
+        :fragment="fragment"
+        @src-loaded="navMenuLoaded"
+      />
+      <div v-else ref="navigationMenu"></div>
+    </div>
+  </div>
+</template>
+
+<script>
+import retriever from './Retriever.vue';
+import $ from './utils/NodeList';
+import { publish, subscribe } from './utils/pubsub';
+
+export default {
+  components: {
+    retriever,
+  },
+  props: {
+    type: {
+      type: String,
+      default: null,
+    },
+    src: {
+      type: String,
+      default: null,
+    },
+    fragment: {
+      type: String,
+      default: null,
+    },
+    hasIdentifier: {
+      type: Boolean,
+      default: false,
+    },
+  },
+  inject: ['toggleLowerNavbar'],
+  data() {
+    return {
+      show: false,
+    };
+  },
+  computed: {
+    buttonId() {
+      switch (this.type) {
+      case 'siteNav':
+        return 'toggle-site-nav-button';
+      case 'pageNav':
+        return 'toggle-page-nav-button';
+      default:
+        return '';
+      }
+    },
+    navMenuIconClass() {
+      switch (this.type) {
+      case 'siteNav':
+        return 'site-nav-menu';
+      case 'pageNav':
+        return 'page-nav-menu';
+      default:
+        return '';
+      }
+    },
+  },
+  methods: {
+    toggleNavMenu() {
+      this.show = !this.show;
+      switch (this.type) {
+      case 'siteNav':
+        publish('closePageNav');
+        break;
+      case 'pageNav':
+        publish('closeSiteNav');
+        break;
+      default:
+        break;
+      }
+    },
+    navMenuLoaded() {
+      this.toggleLowerNavbar();
+      $(this.$refs.navMenuContainer).find('a').on('click', () => {
+        this.toggleNavMenu();
+      });
+    },
+  },
+  mounted() {
+    const navMenu = this.$refs.navigationMenu;
+    const buildNav = (navMenuItems) => {
+      for (let i = 0; i < navMenuItems.childNodes.length; i += 1) {
+        navMenu.appendChild(navMenuItems.childNodes[i].cloneNode(true));
+      }
+      this.navMenuLoaded();
+    };
+    const buildPageNav = () => {
```

Comment:
> Alright I've decided to use <retriever> to get the contents of the identifier id=page/site-nav. However, if these identifiers are not present, will have to look for their respective classes still so it will fallback to using appendChild. Not sure if that's fine?

sounds good, these don't have anything reactive

should we shift these upward though? (into site / page-nav-button) (as a callback accepting `navMenu` maybe)
so the base component only handles generic logic, the specific stuff stays up there

Reply:
Yup sounds good. Shifted this up to `site / page-nav-button`